types/wlr_output.c

```diff
@@ -168,6 +170,28 @@ bool wlr_output_set_cursor(struct wlr_output *output,
 				WL_SHM_FORMAT_ARGB8888, stride, width, height, buf);
 }
 
+bool wlr_output_set_cursor_surface(struct wlr_output *output,
+		struct wlr_surface *surface, int32_t hotspot_x, int32_t hotspot_y) {
+	if (output->impl->set_cursor) {
+		output->impl->set_cursor(output, NULL, 0, 0, 0, 0, 0);
```

Comment:
How to implement this with a `wlr_surface` as input? Some backends like DRM already draw the cursor data to a `wlr_texture`, so converting the `wlr_surface` to a buffer seems redundant in this case.

Reply:
Should be able to by digging into the surface's buffer.